DOCUMENTATION.md

```diff
@@ -0,0 +1,30 @@
+This document describes our rules for documenting things
+
+## The Rules
+
+1. Remember that all our documentation is public-facing.
+2. All engineering documentation in its final state must live in GitHub.
+3. Draft / WIP engineering documentation can live in Google Drive if desired
+4. Code documenation of 3 paragraphs or longer must go into a `.md` file in the same location as the code. You must be able to find every engineering `.md` file via "walking down" links from the root README.md of the repository. Which is to say, add a link from `DEVELOPEMENT.md` or `OPERATIONS.md` or `README.md` (any of them, there are several) into your new `.md` file.
+6. Code comments of 1 paragraph of longer, [or otherwise sufficiently complex](https://github.com/HHS/simpler-grants-gov/blob/02d99c055734edb3d705044de669f7856c0a8fe7/api/tests/src/api/users/test_user_route_login.py#L16-L41), must be linked to from a `.md` file, any `.md` file, pick whichever one makes the most sense.
+7. When documentation in AWS, the written and screenshot documentation must have all sensitive / unique / secrets information redacted, including but not limited to:
+
+  - secrets
+  - unique id's / UUIDs
+  - information that is unique to a given user
+  - anything related to IAM
+
+7. AWS links and screenshots that only contain a generic URL (ex. https://us-east-1.console.aws.amazon.com/rds/home?region=us-east-1#databases) and generic names visible on the screen (eg. simply `api-dev` without any attached unique IDs) are generally safe. If in doubt, don't include it.
+8. Architecture diagrams go into Github. A link into the source drafting tool for the architecture diagram must be added as well, like for example Lucid Chart.
+9. Github issues / PRs documenting durable information of 3 paragraphs of longer, or of sufficient complexity or value, must be linked to from a `.md` file.
+10. Avoid linking into the repo's root README.md file if at all reasonable. 
+11. Always prefer documentation living close to its code, rather than inside of the `documentation/` folder. Tech specs / preliminary design documentation is the exception to this. If in doubt:
+
+  - Documentation created before code is written may go inside of `documentation/`
+  - Documentation created after code is written should not go inside of `documentation/`
```

Comment:
How would you recommend we document something that lives across a few paths?

For example, on the API side there are a few folders:
* /src/api - where we store API endpoints
* /src/services - where we store most of the implementation for a given endpoint (to keep the endpoint logic focused on the input/output/configurational bits)
* /src/auth - where we implement the auth code (used by nearly everything in /src/api)

Auth in particular has a lot built into each one, and also has a lot of implicit behavior with how it might be used - following standard OAuth patterns. 

For this, would it make more sense to document it somewhere in documentation, and then link to it in each of those places?

---

I generally prefer a centralized document hub as putting a ton of it mixed into the code requires knowing where to look. I do think there is a middleground of "document centrally, reference everywhere" to get the best of both worlds.

Reply:
Can you think of some language we can add in here to make this more obvious?

Reply:
That's valid I agree with the general idea but I still want suggestions on how to capture it in this document!

Reply:
Sorry I might not have been clear with my response, but my suggestions would be to replace item 11 with something like:

```
11. Reserve the `documentation/` folder at the root of the repo for engineering docs that apply across multiple codebases, such as onboarding guides, runbooks, and high-level technical overviews.
12. Use a `docs/` folder within each codebase’s root (e.g., `api/docs/`, frontend/docs/) for detailed technical documentation, such as API references, design decisions, and system architecture diagrams.
13. Provide contextual summaries using README.md files in relevant subfolders of the codebase (e.g., api/src/db/README.md). These files should give _short_ overviews of the subfolder's purpose and its key components, but otherwise link to relevant sections in the corresponding `docs/` folder.
```